Ginkgo: Simplify all --focus regexes

jrajahalme opened the pull request:
Ginkgo --focus option is a regular expression that matches a substring
without any trailing wildcarding. Simplify all --focus parameters
accordingly.  The --focus parameter is also not repeatable, the last
one takes the effect if multiple ones are given. Fix the docs for
this.

Update the document to the current default K8S_VERSION (1.18).

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

pchaigno commented on Documentation/contributing/testing/e2e.rst:
```diff
@@ -250,15 +250,17 @@ If you want to run one specified test, there are a few options:
-test that starts with "L7".
+This will focus on test that contains "Runtime", followed by any
+number of any characters, followed by "L7". --focus is a regular
+expression and quotes are required is it contains spaces and to escape
```
```suggestion
expression and quotes are required if it contains spaces and to escape
```

jrajahalme replied:
Fixed.

pchaigno reviewed (approved):
One nit below. LGTM otherwise!

Thanks for fixing this Jarno! :tada: It was really confusing when I started.

tklauser commented on Documentation/contributing/testing/e2e.rst:
```diff
@@ -250,15 +250,17 @@ If you want to run one specified test, there are a few options:
 
-This will focus on tests prefixed with "Run*", and within that focus, run any
-test that starts with "L7".
+This will focus on test that contains "Runtime", followed by any
```
```suggestion
This will focus on tests that contain "Runtime", followed by any
```

jrajahalme replied:
Fixed.

tklauser commented on Documentation/contributing/testing/e2e.rst:
```diff
@@ -250,15 +250,17 @@ If you want to run one specified test, there are a few options:
-This will focus on tests prefixed with "Run*", and within that focus, run any
-test that starts with "L7".
+This will focus on test that contains "Runtime", followed by any
+number of any characters, followed by "L7". --focus is a regular
```
```suggestion
number of any characters, followed by "L7". ``--focus`` is a regular
```

jrajahalme replied:
Fixed.

tklauser commented on Documentation/contributing/testing/e2e.rst:
```diff
@@ -250,15 +250,17 @@ If you want to run one specified test, there are a few options:
+This will focus on test that contains "Runtime", followed by any
+number of any characters, followed by "L7". --focus is a regular
+expression and quotes are required is it contains spaces and to escape
+shell expansion of '*'.
```
```suggestion
shell expansion of ``*``.
```

jrajahalme replied:
Fixed.

tklauser reviewed (approved):
Some small nits, otherwise LGTM.

jrajahalme committed:
Ginkgo: Simplify all --focus regexes

Ginkgo --focus option is a regular expression that matches a substring
without any trailing wildcarding. Simplify all --focus parameters
accordingly.  The --focus parameter is also not repeatable, the last
one takes the effect if multiple ones are given. Fix the docs for
this.

Update the document to the current default K8S_VERSION (1.18).

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
